hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/logical/FullScanLogicalRel.java

```diff
@@ -42,4 +49,28 @@ public FullScanLogicalRel(
     public final RelNode copy(RelTraitSet traitSet, List<RelNode> inputs) {
         return new FullScanLogicalRel(getCluster(), traitSet, getTable(), lagExpression(), watermarkedColumnIndex());
     }
+
+    @Override
+    public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
+        // Prefer scans with smaller number of results/expressions.
+        // This is needed to ensure that UPDATE scans use only key column
+        // TODO: make this calculation consistent with FullScanPhysicalRel cost
```

Comment:
Actually, it doesn't have much sense, since FullScan in logical convention doesn't have an alternatives to be replaced with.

Reply:
but we can full scans with different projections and filters during logical optimization. Also in the future we may plan nested loop join explicitly, in Calcite it starts on logical level (`LogicalCorrelate`) and has different predicates and projections on nested side due to correlation variable being pushed.